compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1098,6 +1103,23 @@ impl<'context> Elaborator<'context> {
         }
     }
 
+    fn set_trait_impl_functions_visibility(&mut self, trait_impl: &UnresolvedTraitImpl) {
+        let trait_visibility = if let Some(trait_id) = trait_impl.trait_id {
+            let trait_ = self.interner.get_trait(trait_id);
+            Some(trait_.visibility)
+        } else {
+            None
+        };
```

Comment:
```suggestion
        let Some(trait_id) = trait_impl.trait_id else { return; };
        let trait_visibility = self.interner.get_trait(trait_id).visibility;
```
Had this as a `map` originally but figured it'd be better to return early instead of go through the next loop if `trait_visibility` is None.